src/main/java/org/jabref/logic/importer/fetcher/ZbMATH.java

```diff
@@ -40,17 +46,55 @@ public String getName() {
         return "zbMATH";
     }
 
-    /**
-     * TODO: Implement EntryBasedParserFetcher
-     * We use the zbMATH Citation matcher (https://www.zbmath.org/citationmatching/)
-     * instead of the usual search since this tool is optimized for finding a publication based on partial information.
-     */
-    /*
     @Override
     public URL getURLForEntry(BibEntry entry) throws URISyntaxException, MalformedURLException, FetcherException {
-        // Example: https://zbmath.org/citationmatching/match?q=Ratiu
+        Optional<String> zblidInEntry = entry.getField(StandardField.ZBL_NUMBER);
+        if (zblidInEntry.isPresent()) {
+            // zbmath id is already present
+            return getUrlForIdentifier(zblidInEntry.get());
+        }
+
+        URIBuilder uriBuilder = new URIBuilder("https://zbmath.org/citationmatching/match");
+        uriBuilder.addParameter("n", "1"); // return only the best matching entry
+        uriBuilder.addParameter("m", "5"); // return only entries with a score of at least 5
+
+        entry.getFieldOrAlias(StandardField.TITLE).ifPresent(title -> uriBuilder.addParameter("t", title));
+        entry.getFieldOrAlias(StandardField.JOURNAL).ifPresent(journal -> uriBuilder.addParameter("j", journal));
+        entry.getFieldOrAlias(StandardField.YEAR).ifPresent(year -> uriBuilder.addParameter("y", year));
+        entry.getFieldOrAlias(StandardField.PAGINATION)
+             .ifPresent(pagination -> uriBuilder.addParameter("p", pagination));
+        entry.getFieldOrAlias(StandardField.VOLUME).ifPresent(volume -> uriBuilder.addParameter("v", volume));
+        entry.getFieldOrAlias(StandardField.ISSUE).ifPresent(issue -> uriBuilder.addParameter("i", issue));
+
+        if (entry.getFieldOrAlias(StandardField.AUTHOR).isPresent()) {
+            // replace "and" by ";" as citation matching API uses ";" for separation
+            String author = entry.getFieldOrAlias(StandardField.AUTHOR).get();
+            author = author.replace(" and ", ";");
+            uriBuilder.addParameter("a", author);
+        }
+
+        String urlString = uriBuilder.build().toString();
+        HttpResponse<JsonNode> response = Unirest.get(urlString)
```

Comment:
Can you please add a comment that the API does not provide a means to return bibtex directly, and we thus extract the zbl id and use getUrlForIdentifier which does return bibtex. 

Btw, could you please contact zbl and maybe ask if they could add a `format=bibtex` query parameter to their match api? That would make this additional roundtrip unnecessary. Thanks!

Reply:
We work on a new citation matching algorithm and we will add bibtex as an output format there. I will change the jabref code if this happens. Right now, I guess it is okay to use this workaround.